test/components/CivicProfileForms/BasicInfo.test.jsx

```diff
@@ -1,83 +1,113 @@
+      legalGender: 1
+    };
+    useCivicProfile.mockReturnValue({
+      add: mockClear,
+      isSuccess: true,
+      storedDataset: {},
+      refetch: vi.fn()
+    });
+    useNotification.mockReturnValue({ addNotification: vi.fn() });
+
+    const { getByRole } = render(<BasicInfo />);
+    const legalFirstName = getByRole('textbox', { name: 'Legal first name' });
+    const legalLastName = getByRole('textbox', { name: 'Legal last name' });
+    const legalDOB = getByRole('textbox', { name: 'Choose date' });
+    const legalGender = getByRole('combobox', { name: 'Gender' });
+    const clearButton = getByRole('button', { name: 'Clear button' });
 
-  //   await userEvent.click(clearButton);
+    await user.type(legalFirstName, basicInfoProfile.legalFirstName);
+    await user.type(legalLastName, basicInfoProfile.legalLastName);
+    await user.type(legalDOB, basicInfoProfile.legalDOB);
+    await user.click(legalGender);
+    const listBox = within(getByRole('listbox'));
+    user.click(listBox.getByText('Female'));
+    await user.click(clearButton);
 
-  //   expect(legalFirstName.value).toBe('');
-  //   expect(legalLastName.value).toBe('');
-  //   expect(legalDOB.value).toBe(null);
-  //   expect(legalGender.value).toBe('');
-  // });
+    expect(legalFirstName.value).toBe('');
+    expect(legalLastName.value).toBe('');
+    expect(legalDOB.value).toBe('');
+    expect(await screen.findByText(/Decline to answer/i)).toBeInTheDocument();
+  });
 
   // TODO: Resolve test not passing
-  // it('submits a basic info profile update when you click the submit button', async () => {
-  //   const user = userEvent.setup();
-  //   const mockAdd = vi.fn();
-  //   const basicInfoProfile = {
-  //     legalFirstName: 'Jane',
-  //     legalLastName: 'Doe',
-  //     legalDOB: '1980-12-15',
-  //     legalGender: 1
-  //   };
-  //   useCivicProfile.mockReturnValue({ add: mockAdd, isSuccess: true });
-  //   const { getByRole } = render(<BasicInfo />);
-  //   const legalFirstName = getByRole('textbox', { name: 'Legal first name' });
-  //   const legalLastName = getByRole('textbox', { name: 'Legal last name' });
-  //   const legalDOB = getByRole('textbox', { name: 'Choose date' });
-  //   const legalGender = getByRole('combobox', { name: 'Gender' });
-  //   const submitButton = getByRole('button', { name: 'Submit button' });
-  //   await user.type(legalFirstName, basicInfoProfile.legalFirstName);
-  //   await user.type(legalLastName, basicInfoProfile.legalLastName);
-  //   await user.type(legalDOB, basicInfoProfile.legalDOB);
-  //   await user.type(legalGender, `${basicInfoProfile.legalGender}{enter}`);
-  //   await user.click(submitButton);
-  //   expect(mockAdd).toBeCalledWith(basicInfoProfile);
-  // });
+  it('submits a basic info profile update when you click the submit button', async () => {
+    const user = userEvent.setup();
+    const mockAdd = vi.fn();
+    const basicInfoProfile = {
+      firstName: 'Jane',
```

Comment:
The basicInfoProfile in this test is set up different than the one in the other. Is there a particular reason for this?

I'm not sure it really matters. Just curious. If the tests pass as they should then that's the important thing.

Reply:
In the first test, the value of the DOB doesn't matter, it just needs to be /something/ so I didn't change it. The 2nd test needs the DOB value (after conversion) to match the value stored by the DatePicker so I had to do a little manipulation and specifically selected the current date. I think I'll change the first test to also select current date so it's not too confusing in the future.